### DISCUSSION TITLE
How to conditionally include a config file?

### DISCUSSION THREAD

**Discussion by johnnyutahh** [helix-editor/helix#8317] [Q&A] [answered]
How to conditionally include a config file?  Is it feasible?

eg, in `~/.config/helix/config.toml` (on Linux or macOS):

```
if [systemA] then source configA.toml
if [systemB] then source configB.toml
```

**Comment by kirawi** [comment]
You can pass `hx -c <config file>` if that's what you're looking for.

**Comment by johnnyutahh** [comment]
> You can pass `hx -c <config file>` if that's what you're looking for.

That's nice to know, thank you, but it does not solve my problem.

Rather, like `vim`, I want to put content in the config file(s) (`~/.config/helix/config.toml`... and/or any other config files) to enable conditional loading of _other_ config files. 

I want to enable similar functionality to vim. Here's an example `.vimrc` snippet that checks `has("gui_macvim")` to see if it should include (`source`) another config file (in reality, this is bad vim/macvim/vim-gui configuration stuff; it's simply an example to show the `if`-plus-`source` feature/functionality of `vim` that I'd like to incorporate into `hx`):

```
if has("gui_macvim")
  source '~/.config/vim/a_macvim_config_file'
endif
```

I'd _also_ like to be able to check conditions like shell-environment variables to determine which config files to include/source/load. Like this (imagine this as `hx` `config.toml` pseudo code):

```
if shellenv_exists('HX_MIMIC_VIM')
{
    source 'helix_config_mimic_vim.toml'
}
```

**Reply by pascalkuthe** [reply, marked as answer]
This won't be possible with the toml config. Once we have a plugin system something like.that will.be possible until then you will habe to use tricks like symlinking a different config file or per project configuration (.helix/config.toml in workspace root)

**Reply by johnnyutahh** [reply]
Got it, thanks much.
